distributed/tests/test_worker.py

```diff
@@ -2522,14 +2535,31 @@ def fast_on_a(lock):
             while f3.key not in a.tasks:
                 await asyncio.sleep(0.01)
 
+            story_before = s.story(f1.key)
             await s.stream_comms[a.address].close()
+            # Release f1 which triggers a release cycle of all tasks such that
+            # they are rescheduled on B. However, at this time, B will never be
+            # able to compute f3 / fast_on_a since it is locked on that worker.
+            # The only way to get f3 to complete is for Worker A to reconnect.
+
             f1.release()
             assert len(s.workers) == 1
-            while s.tasks[f1.key].state != "released":
-                await asyncio.sleep(0)
+            story = s.story(f1.key)
+            while len(story) == len(story_before):
+                story = s.story(f1.key)
+                await asyncio.sleep(0.1)
+
+            next = story[len(story_before)]
+            assert next[0] == f1.key
+            assert next[1] == "memory"
+            assert next[2] == "released"
```

Comment:
```suggestion
            story_after = story[len(story_before):]
            assert_worker_story(
                story_after,
                [
                    (f1.key, "memory", "released"),
                 ],
             )
```
This might need slight tweaking if there's more stuff on the right of the event.
I'd encourage to add strict=True and assert the full contents of the story, if possible.

Reply:
We're dealing with scheduler stories here. The assert_worker_story does not work for the scheduler, yet, and I netiher wanted to write an `assert_scheduler_story` nor migrate the scheduler to stimulus IDs in this PR

Reply:
Ah I missed this.